File: ui/src/ui/sample/SampleRequest.ts
import Coordinate from "./Coordinate";
import SampleLayer from "./dto/SampleLayer";
import SampleObject from "./dto/SampleObjects";
import SampleSize from "./dto/SampleSize";

class SampleRequest {
    coordinate: Coordinate;
    objects: SampleObject[];
    size: SampleSize;
    layer: SampleLayer;
}

File: src/main/java/com/djulb/ui/sample/SampleRequest.java
package com.djulb.ui.sample;

import com.djulb.ui.sample.dto.SampleLayer;
import com.djulb.ui.sample.dto.SampleObject;
import com.djulb.ui.sample.dto.SampleSize;
import com.djulb.way.bojan.Coordinate;
import lombok.Data;

import java.util.List;

@Data
public class SampleRequest {
    double lat;
    double lng;
    List<SampleObject> objects;
    SampleSize size;
    SampleLayer layer;

    public Coordinate getCoordinate() {
        return Coordinate.builder().lat(lat).lng(lng).build();
    }
}
